CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKRenderer.java

```diff
@@ -133,40 +135,52 @@ private void DestroyImplementation() {
 
         if (cvkDevice != null) {
             cvkDevice.Destroy();
-            cvkDevice = null;        
         }
     }
-    
-    
+        
     /**
-     * Called from the parent CVKVisualProcessor when the VisualManager is
-     * being destroyed.
+     * Called from CVKCanvas when it's being destroyed.
      */
     public void Destroy() {      
         isExiting = true;
-        parent.cvkLogger.info("CVKRenderer.destroy() called");
-        if (parent.IsRenderThreadCurrent()) {           
+        GetLogger().info("CVKRenderer.destroy() called");
+        if (cvkCanvas.IsRenderThreadCurrent()) {           
             if (cvkDevice != null) {
                 cvkDevice.WaitIdle();
             }
 
             DestroyImplementation();
         } else {
-            if (parent.IsRenderThreadAlive()) {
+            if (cvkCanvas.IsRenderThreadAlive()) {
```

Comment:
Can this be false ?

Reply:
Possibly....I moved it from being called by the visual processor (which is operated on by different threads) and the canvas.  Will leave for now.